fairscale/experimental/wgit/weigit_api.py

```diff
@@ -0,0 +1,172 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+import json
+import os
+import pathlib
+from pathlib import Path
+import shutil
+import sys
+from typing import Union
+
+import pygit2
+
+from fairscale.experimental.wgit.pygit import PyGit
+from fairscale.experimental.wgit.sha1_store import SHA1_store
+from fairscale.experimental.wgit.utils import ExitCode, create_dir, create_file, get_sha1_hash, write_to_file
+
+
+class WeiGitRepo:
+    """
+    Features:
+    1. Create the wgit directory if it does not exist.
+    2. SHA1Store.init()
+    3. Create SHA1 .wgit/sha1_refs.json
+    3. Initialize a .git directory within the .wgit using `git init`.
+    4. add a .gitignore within the .wgit directory, so that the git repo within will ignore `sha1_refs.json`
+    """
+
+    def __init__(self, parent_dir: Path, init: bool = False) -> None:
+        # If repo does not exist, creates a new wgit repo object with self.repo.path pointing to the path of repo
+        # and notes all the internal files.
+        # else, if repo already exists: create a pygit object from the .wgit/.git.
+        self.wgit_parent = parent_dir
+        self._repo_path: Union[None, Path] = None
+        self._wgit_dir = Path(".wgit")
+        self._metadata_file = Path(".wgit/checkpoint.pt")
+        self._sha1_ref = Path(".wgit/sha1_refs.json")
+        self._wgit_git_path = Path(".wgit/.git")
+        self._sha1_store = Path(".wgit/sha1_store")
+
+        if not self._exists() and init:
+            # No weigit repo exists and is being initialized with init=True
+            # Make .wgit directory, create sha1_refs and metadata file
+            create_dir(dir_path=self._wgit_dir, exception_msg="An exception occured: WeiGit already Initialized")
+            create_file(file_path=self._metadata_file)
+            create_file(file_path=self._sha1_ref)
+
+            # Make the .wgit a git repo
+            try:
+                pygit2.init_repository(str(self._wgit_git_path), False)
```

Comment:
This probably should be part of the PyGit class and should handle self.user and self.email's init.